src/main/java/org/asamk/signal/util/Util.java

```diff
@@ -21,7 +21,7 @@ public static String getSecret(int size) {
 
     private static SecureRandom getSecureRandom() {
         try {
-            return SecureRandom.getInstance("SHA1PRNG");
+            return SecureRandom.getInstance("NativePRNG");
```

Comment:
May I recommend simply using `new SecureRandom()` instead? As per the documentation, on Solaris, Linux, and OS X, this is (currently) equivalent to `SecureRandom.getInstance("NativePRNG")`. However, on other platforms (such as Windows) you'll have problems as "NativePRNG" might not exist.  In all cases just using `new SecureRandom()` will give you what the JVM believes to be the best for your current system. This means that as the JVM is continually improved and developed, you will automatically get improvements in your random number generation.  (It also makes it easier for people to swap out implementations if they have other security requirements, such as running on a FIPS validated provider.)

A few other things to consider:
1. Immediately after creation, it's best for force seeding by calling `.nextBoolean()`. This keeps you safe if a consumer incorrectly calls `.setSeed()` as only the initial seeding has security concerns.
2. Entropy generation can be a thread-contention problem on many systems.  If you are worried about this, I recommend using a `ThreadLocal` construction.

Both of these things are how we do it in [some of our other projects](https://github.com/aws/aws-encryption-sdk-java/blob/master/src/main/java/com/amazonaws/encryptionsdk/internal/Utils.java#L32).

Reply:
Good input. But as far as I can tell this project is (right now) not aimed at Windows systems at all and there doesn't seem to be a possibility to use it on Windows nor are there and users I've read of.

`ThreadLocal`: good input! You should discuss this with the original author.

Reply:
Wasn't aware, ok thanks, I was planning on chaning to `new SecureRandom` anyhow because of @SalusaSecondus comment.

Reply:
I'm wondering if there's a reason we should force seeding with `.nextBoolean()` now as I looked at the AWS code mentioned by @SalusaSecondus. Hm. I'm not sure if in this project there're any consumers that would be able to incorrectly call `.setSeed()` - it would imply the project itself being malicious. it's not a general purpose library after all.